pkg/ingester/ingester.go

```diff
@@ -1491,11 +1505,11 @@ func (i *Ingester) createTSDB(userID string) (*userTSDB, error) {
 		IsolationDisabled:              !i.cfg.BlocksStorageConfig.TSDB.IsolationEnabled,
 		HeadChunksWriteQueueSize:       i.cfg.BlocksStorageConfig.TSDB.HeadChunksWriteQueueSize,
 		NewChunkDiskMapper:             i.cfg.BlocksStorageConfig.TSDB.NewChunkDiskMapper,
-		AllowOverlappingQueries:        i.cfg.BlocksStorageConfig.TSDB.OOOAllowance > 0, // always true if out of order support is enabled
-		AllowOverlappingCompaction:     false,                                           // always false since Mimir only uploads lvl 1 compacted blocks
-		OOOAllowance:                   int64(i.cfg.BlocksStorageConfig.TSDB.OOOAllowance / time.Millisecond),
-		OOOCapMin:                      int64(i.cfg.BlocksStorageConfig.TSDB.OOOCapMin),
-		OOOCapMax:                      int64(i.cfg.BlocksStorageConfig.TSDB.OOOCapMax),
+		AllowOverlappingQueries:        oooAllowance > 0,            // true if out of order support is enabled
```

Comment:
Since `oooAllowance` can change from 0 to positive value in runtime, shall we ALWAYS enable querying of overlapping blocks? (In that case let's also update description for oooAllowance limit).

Reply:
Oh right. We should always enable yes. Because after a restart if you disabled ooo, you will still have overlapping blocks. That said, we seem to have a config flag for this in the main branch https://github.com/grafana/mimir/blob/bee820659c179617c75f2b12a900ae1fc91545ab/pkg/ingester/ingester.go#L1476 (I thought it was always set to true). I can change it in the main branch as well to always set it to true and get rid of that unnecessary flag. WDYT?

Reply:
I can make the change in main as part of merging out of order work in main.